From ghaskins@novell.com Thu Jun  5 09:51:54 2008
Date: Wed, 04 Jun 2008 15:04:10 -0400
From: Gregory Haskins <ghaskins@novell.com>
To: Ingo Molnar <mingo@elte.hu>, Thomas Gleixner <tglx@linutonix.de>,
     Steven Rostedt <rostedt@goodmis.org>,
     Arnaldo Carvalho de Melo <acme@redhat.com>
Cc: Gregory Haskins <ghaskins@novell.com>, linux-kernel@vger.kernel.org,
     linux-rt-users@vger.kernel.org, Peter Zijlstra <peterz@infradead.org>
Subject: [PATCH 2/2] sched: fix cpupri priocount

A rounding error was pointed out by Peter Zijlstra which would result
in the structure holding priorities to be off by one.

Signed-off-by: Gregory Haskins <ghaskins@novell.com>
CC: Peter Zijlstra <peterz@infradead.org>

Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>

---

 kernel/sched_cpupri.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.25.4-rt6/kernel/sched_cpupri.h
===================================================================
--- linux-2.6.25.4-rt6.orig/kernel/sched_cpupri.h	2008-06-05 23:39:47.000000000 -0400
+++ linux-2.6.25.4-rt6/kernel/sched_cpupri.h	2008-06-05 23:40:45.000000000 -0400
@@ -4,7 +4,7 @@
 #include <linux/sched.h>
 
 #define CPUPRI_NR_PRIORITIES 2+MAX_RT_PRIO
-#define CPUPRI_NR_PRI_WORDS CPUPRI_NR_PRIORITIES/BITS_PER_LONG
+#define CPUPRI_NR_PRI_WORDS (CPUPRI_NR_PRIORITIES + BITS_PER_LONG/2)/BITS_PER_LONG
 
 #define CPUPRI_INVALID -1
 #define CPUPRI_IDLE     0
